### DISCUSSION TITLE
Select2 component won't work when using selection.

### DISCUSSION THREAD

**Discussion by desyashasyi** [livewire/livewire#5437] [1. Help]
I expect to have a select2 component in the livewire child component.

When I composed the blade structure like this, the Select2 rendered it properly.

`
...

 <div class="card">
        @livewire('arsys.admin.user.staffs')
</div>

...

`
![image](https://user-images.githubusercontent.com/48233515/207200755-e39782c4-d733-4404-8740-093f7d66243d.png)

But when I composed the Livewire page like this, the Select2 component does not render

`
...

<div class="card">
        @if(!is_null($selectedMenu))
            <div class="card-body">
               
                @if($selectedMenu == 'staff')
                    @livewire('arsys.admin.user.staffs')
                @endif
               
            </div>
        @endif
    </div>
...
`

![image](https://user-images.githubusercontent.com/48233515/207200834-4b60aa93-5fc8-4cae-b331-c7b0cf411b2a.png)

The parent controller

`
<?php

namespace App\Http\Livewire\Arsys\Admin\User;

use Livewire\Component;

class Selector extends Component
{
    public $selectedMenu = null;
    public function render()
    {
        return view('livewire.arsys.admin.user.selector');
    }

   
    public function selectMenu($menu){
        if(!is_null($menu)){
            $this->selectedMenu = $menu;

            if($menu == 'account'){
                $this->emit('accountActive');
            }
            if($menu == 'staff'){
                $this->emit('staffActive');
            }
            if($menu == 'student'){
                $this->emit('studentActive');
            }
        }
    }
}

`

The parent blade

`
<div>
    
    <ul class="nav nav-tabs" id="myTab" role="tablist">
        <li class="nav-item">
          <a class="nav-link" id="account-tab" href="#" style="cursor:pointer" wire:click="selectMenu('account')" data-toggle="tab" role="tab" aria-controls="account" aria-selected="true">Account</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="staff-tab" href="#" style="cursor:pointer" wire:click="selectMenu('staff')" data-toggle="tab"  role="tab" aria-controls="staff" aria-selected="false">Staff</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="student-tab" href="#" style="cursor:pointer" wire:click="selectMenu('student')" data-toggle="tab" role="tab" aria-controls="student" aria-selected="false">Student</a>
        </li>
    </ul>
    <div class="card">
        @if(!is_null($selectedMenu))
            <div class="card-body">
                @if($selectedMenu == 'account')
                    @livewire('arsys.admin.user.account')
                @endif
                @if($selectedMenu == 'staff')
                    @livewire('arsys.admin.user.staffs')
                @endif
                @if($selectedMenu == 'student')
                    @livewire('arsys.admin.user.students')
                @endif
            </div>
        @endif
    </div>
    @push('script')
    <script>
        $(document).ready(function(){
            window.livewire.on('accountActive',()=>{
                $('#account').addClass('active');
                $('#staff').removeClass('active');
                $('#student').removeClass('active');
                
            });
            window.livewire.on('staffActive',()=>{
                $('#staff').addClass('active');
                $('#account').removeClass('active');
                $('#student').removeClass('active');
            });
            window.livewire.on('studentActive',()=>{
                $('#student').addClass('active');
                $('#staff').removeClass('active');
                $('#account').removeClass('active');
            });
            
    }); 
    </script>
    @endpush
</div>

`

The child controller

`
<?php

namespace App\Http\Livewire\Arsys\Admin\User;

use Livewire\Component;
use Livewire\WithPagination;
use App\Models\ArSys\Staff;

class Staffs extends Component
{
    public $enabledExpandView = [];
    use WithPagination;
    public $viewIndex = null;
    public $tempIndex = null;
    public $addStaffEnable = null;
    public $viewStaffEnable = null;
    public $viewEnable = null;
    public $menu;
    protected $listeners = ['closeStaffsView_AdminUserStaffView', 'selfRefresh' => '$refresh', 'addStaffEnable_AdminUserStaff'];
    protected $paginationTheme = 'bootstrap';
    public $pageNumber = null;
    public function render()
    {
        if($this->tempIndex != $this->viewIndex){
            $this->enabledExpandView[$this->viewIndex] = 1;
            $this->enabledExpandView[$this->tempIndex] = null;
            $this->tempIndex = $this->viewIndex;
        }     
        $staffs = Staff::where('position_id', '!=', null)->orderBy('position_id', 'ASC')
        ->orderBy('univ_code', 'ASC')->paginate(10);
        
        if($this->pageNumber != $staffs->currentPage()){
            foreach($staffs as $index => $staff){
                $this->enabledExpandView[$index] = null;
            }
            $this->pageNumber = $staffs->currentPage();
            $this->tempIndex = $staffs->count()+1;
            $this->viewIndex = $staffs->count()+1;
        }
        return view('livewire.arsys.admin.user.staffs', ['staffs' => $staffs]);
    }

    public function expandView($viewIndex, $staffId){
        if(is_null($this->viewStaffEnable)){
            $this->viewStaffEnable = 1;
        }else{
            $this->viewStaffEnable = null;
        }
        if($this->tempIndex != $viewIndex){
            $this->viewIndex = $viewIndex;
            $this->emit('userStaffView_AdminUserStaff', ['StaffId' => $staffId]);
        }else{
            if($this->enabledExpandView[$this->tempIndex] == null){
                $this->enabledExpandView[$this->tempIndex] = 1;
                $this->emit('userStaffView_AdminUserStaff', ['StaffId' => $staffId]);
            }else{
                $this->enabledExpandView[$this->tempIndex] = null;
            }
        }
    }

    public function closeStaffsView_AdminUserStaffView(){
        foreach($this->staffs as $index => $staff){
            $this->enabledExpandView[$index] = null;
        }
    }

    public function addStaffEnable_AdminUserStaff(){
        if(is_null($this->addStaffEnable)){
            $this->addStaffEnable = 1;
        }else{
            $this->addStaffEnable = null;
        }
    }
}

`

The child blade

`
<div>
    
    <ul class="nav nav-tabs" id="myTab" role="tablist">
        <li class="nav-item">
          <a class="nav-link" id="account-tab" href="#" style="cursor:pointer" wire:click="selectMenu('account')" data-toggle="tab" role="tab" aria-controls="account" aria-selected="true">Account</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="staff-tab" href="#" style="cursor:pointer" wire:click="selectMenu('staff')" data-toggle="tab"  role="tab" aria-controls="staff" aria-selected="false">Staff</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="student-tab" href="#" style="cursor:pointer" wire:click="selectMenu('student')" data-toggle="tab" role="tab" aria-controls="student" aria-selected="false">Student</a>
        </li>
    </ul>
    <div class="card">
        @if(!is_null($selectedMenu))
            <div class="card-body">
               
                @if($selectedMenu == 'staff')
                    @livewire('arsys.admin.user.staffs')
                @endif
               
            </div>
        @endif
    </div>
    @push('script')
    <script>
        $(document).ready(function(){
            window.livewire.on('accountActive',()=>{
                $('#account').addClass('active');
                $('#staff').removeClass('active');
                $('#student').removeClass('active');
                
            });
            window.livewire.on('staffActive',()=>{
                $('#staff').addClass('active');
                $('#account').removeClass('active');
                $('#student').removeClass('active');
            });
            window.livewire.on('studentActive',()=>{
                $('#student').addClass('active');
                $('#staff').removeClass('active');
                $('#account').removeClass('active');
            });
            
    }); 
    </script>
    @endpush
</div>

`

Any advice?

**Comment by joshhanley** [comment]
@desyashasyi your code formatting seems to not be loading properly, which is making it hard to follow. Can you fix it up please?

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\ PHP code here
\```

\```blade
\ HTML and blade code here
\```

Thanks!
